packages/react-bootstrap-table2/test/bootstrap-table.test.js

```diff
@@ -120,4 +120,109 @@ describe('BootstrapTable', () => {
       expect(body.props().cellEdit.onComplete).toBeDefined();
     });
   });
+
+  describe('handleRowSelect', () => {
+    const rowKey = 1;
+
+    describe('when selectRow.mode is radio', () => {
+      beforeEach(() => {
+        wrapper = shallow(
+          <BootstrapTable
+            keyField="id"
+            columns={ columns }
+            data={ data }
+            selectRow={{ mode: 'radio' }}
+          />
+        );
+      });
+
+      it('state.selectedRowKeys should contain single key', () => {
+        wrapper.instance().handleRowSelect(rowKey);
+
+        expect(wrapper.state('selectedRowKeys')).toEqual([rowKey]);
+      });
+    });
+
+    describe('when selectRow.mode is checbox', () => {
+      beforeEach(() => {
+        wrapper = shallow(
+          <BootstrapTable
+            keyField="id"
+            columns={ columns }
+            data={ data }
+            selectRow={{ mode: 'checkbox' }}
+          />
+        );
+      });
+      describe('if checked is false', () => {
+        it('state.selectedRowKeys should pop selected row key', () => {
+          wrapper.instance().handleRowSelect(rowKey, false);
+
+          expect(wrapper.state('selectedRowKeys')).not.toContain(rowKey);
+        });
+      });
+
+      describe('if checked is true', () => {
+        it('state.selectedRowKeys should push one extra key', () => {
+          wrapper.instance().handleRowSelect(rowKey, true);
+
+          expect(wrapper.state('selectedRowKeys')).toContain(rowKey);
+        });
+      });
+    });
+  });
+
+  describe('handleAllRowsSelect', () => {
+    beforeEach(() => {
+      wrapper = shallow(
+        <BootstrapTable
+          keyField="id"
+          columns={ columns }
+          data={ data }
+        />
+      );
+    });
+
+    describe('when customized option was not given', () => {
+      describe('when nothing was selected', () => {
+        it('should select all rows', () => {
+          wrapper.setState({ selectedRowKeys: [] });
+
+          wrapper.instance().handleAllRowsSelect();
+
+          expect(wrapper.state('selectedRowKeys').length).toBe(data.length);
+        });
+      });
+
+      describe('when one or more than one row was selected', () => {
+        it('should unselect all rows', () => {
+          wrapper.setState({ selectedRowKeys: [...data] });
```

Comment:
`data` is an object array.. give some fake data is ok:  [ 1, 3 ,4 ].
My concern is `selectedRowKeys` is key of array instead of object